### PR TITLE
Co owner authorizations and email

### PR DESCRIPTION
**Pull request by TCW417**
Added authorization controls to project volunteer cards based on user being creator or co-owner of project.
Added co-owners emails to Contact Project email.

### CODE REVIEW

File: civictechprojects/views.py [hunk 1]
```diff
@@ -347,6 +350,7 @@ def accept_project_volunteer(request, application_id):
 @csrf_exempt
 def promote_project_volunteer(request, application_id):
     volunteer_relation = VolunteerRelation.objects.get(id=application_id)
+    print('promote_project_volunteer, request username', request.user.username, 'creator:', volunteer_relation.project.project_creator.username)
```
Comment (marlonkeating) [inline civictechprojects/views.py]: Leftover logging?

File: common/components/utils/CurrentUser.js [hunk 2]
```diff
@@ -25,6 +27,13 @@ class CurrentUser {
   static isStaff() : boolean {
     return window.DLAB_GLOBAL_CONTEXT.isStaff;
   }
+
+  static isCoOwner(project: ProjectDetailsAPIData) {
```
Comment (marlonkeating) [inline common/components/utils/CurrentUser.js]:
```suggestion
  static isCoOwner(project: ProjectDetailsAPIData) : boolean {
```

Comment (marlonkeating) [inline common/components/utils/CurrentUser.js]: Make sure to document the fact that the project creator does _not_ count as a co-owner.

File: common/components/common/volunteers/VolunteerSection.jsx [hunk 3]
```diff
@@ -295,6 +296,7 @@ class VolunteerSection extends React.PureComponent<Props, State> {
                     key={i}
                     volunteer={volunteer}
                     isProjectAdmin={this.props.isProjectAdmin}
+                    isProjectCoOwner={this.props.isProjectCoOwner && !(header === "CO-OWNERS")}
```
Comment (marlonkeating) [inline common/components/common/volunteers/VolunteerSection.jsx]: "CO-OWNERS" should be a constant if we're using it in multiple places

File: civictechprojects/views.py (hunk 1 above)
Comment (TCW417) [reply civictechprojects/views.py]: Yes. Oops. Missed it.  Deleted.

File: common/components/utils/CurrentUser.js (hunk 2 above)
Comment (TCW417) [reply common/components/utils/CurrentUser.js]: Done.

Comment (TCW417) [reply common/components/utils/CurrentUser.js]: Added comments to currentuser.js and volunteersection.jsx

File: common/components/common/volunteers/VolunteerSection.jsx (hunk 3 above)
Comment (TCW417) [reply common/components/common/volunteers/VolunteerSection.jsx]: We're using it 2 times within a few lines of each other. Does that rate adding it to ProjectConstants?

File: civictechprojects/views.py [hunk 4]
```diff
@@ -255,6 +255,13 @@ def presign_project_thumbnail_upload(request):
         raw_key=s3_key, file_name=file_name, file_type=file_type, acl="public-read")
 
 
+def request_is_authorized(request, volunteer_relation):
```
Comment (marlonkeating) [inline civictechprojects/views.py]:
```suggestion
def volunteer_operation_is_authorized(request, volunteer_relation):
```

File: common/components/common/volunteers/VolunteerSection.jsx (hunk 3 above)
Comment (TCW417) [reply common/components/common/volunteers/VolunteerSection.jsx]: Got it.  Done.

File: civictechprojects/views.py (hunk 4 above)
Comment (TCW417) [reply civictechprojects/views.py]: Done.

File: common/components/constants/ProjectConstants.js [hunk 5]
```diff
@@ -10,4 +10,6 @@ export const Locations = {
     "Baltimore, MD",
     "Other"
   ]
-};
\ No newline at end of file
+};
+
+export const CoOwnerHeading = "CO-OWNERS"
```
Comment (marlonkeating) [inline common/components/constants/ProjectConstants.js]: We might as well put this in VolunteerSection.jsx since it isn't used outside of that file

Comment (TCW417) [reply common/components/constants/ProjectConstants.js]: Done.  Put it at the top after the imports. Or should it be in class scope?

Comment (marlonkeating) [reply common/components/constants/ProjectConstants.js]: That's a good spot for it, I think.

**Pull request merged**